pyccel/codegen/printing/cwrappercode.py

```diff
@@ -159,7 +198,170 @@ def get_cast_function_call(self, cast_type, arg):
+                    Print TypeError Wrong Type
+                    return Null
+                }else{
+                    collect the value from PyObject
+                }
+        Parameters:
+        ----------
+        Variable : Variable
+            The optional variable
+        collect_var : variable
+            the pyobject type variable  holder of value
+        check_type : Boolean
+            True if the type is needed
+
+        Returns
+        -------
+        body : list
+            A list of statements
+        """
+        body = [(PyccelEq(VariableAddress(collect_var), VariableAddress(Py_None)),
+                [Assign(variable, variable.value)])]
+        if check_type : # Type check
+            check = PyccelNot(PythonType_Check(variable, collect_var))
+            error = PyErr_SetString('PyExc_TypeError', '"{} must be {}"'.format(variable, variable.dtype))
+            body += [(check, [error, Return([Nil()])])]
+        body += [(LiteralTrue(), [Assign(variable, self.get_collect_function_call(variable, collect_var))])]
+        body = [If(*body)]
+
+        return body
+
+    def _body_array(self, variable, collect_var, check_type = False) :
+        """
+        Responsible for collecting value and managing error and create the body
+        of arguments with rank greater than 0 in format
+                if (rank check == False){
+                    print TypeError Wrong rank
+                    return Null
+                }else if(Type Check == False){
+                    Print TypeError Wrong type
+                    return Null
+                }else if (order check == False){ #check for order for rank > 1
+                    Print NotImplementedError Wrong Order
+                    return Null
+                }
+                collect the value from PyArrayObject
+
+        Parameters:
+        ----------
+        Variable : Variable
+            The optional variable
+        collect_var : variable
+            the pyobject type variable  holder of value
+        check_type : Boolean
+            True if the type is needed
+
+        Returns
+        -------
+        body : list
+            A list of statements
+        """
+        #TODO create and extern rank and order check function
+        #rank check :
+        check = PyccelNe(FunctionCall(numpy_get_ndims,[collect_var]), LiteralInteger(collect_var.rank))
+        error = PyErr_SetString('PyExc_TypeError', '"{} must have rank {}"'.format(collect_var, str(collect_var.rank)))
+        body  = [(check, [error, Return([Nil()])])]
+
+        if check_type : #Type check
+            numpy_dtype = self.find_in_numpy_dtype_registry(variable)
+            arg_dtype   = self.find_in_dtype_registry(self._print(variable.dtype), variable.precision)
+            check = PyccelNe(FunctionCall(numpy_get_type, [collect_var]), numpy_dtype)
+            info_dump = PythonPrint([FunctionCall(numpy_get_type, [collect_var]), numpy_dtype])
```

Comment:
Did I leave the `info_dump` in the master branch?! Oops^^

Reply:
Although, maybe some of this is useful. It would make sense to put it in SetString but we can't do that yes

Reply:
I was thinking the same but I don't think we can raise 2 errors, maybe we can change error code format

Reply:
Maybe just leave it as it is until strings in c are more advanced